test/groovy/DockerExecuteTest.groovy

```diff
@@ -1,61 +1,97 @@
 
 import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
 
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import util.JenkinsConfigRule
+import util.JenkinsLoggingRule
+import util.JenkinsSetupRule
+
 import static org.junit.Assert.assertEquals
 import static org.junit.Assert.assertTrue
 import static org.junit.Assert.assertFalse
 
-class DockerExecuteTest extends PiperTestBase {
+class DockerExecuteTest extends BasePipelineTest {
     private DockerMock docker
 
-    String echos
+    @Rule
+    public JenkinsSetupRule jsr = new JenkinsSetupRule(this)
+
+    @Rule
+    public JenkinsLoggingRule jlr = new JenkinsLoggingRule(this)
+
+    @Rule
+    public JenkinsConfigRule jcr = new JenkinsConfigRule(this)
 
     int whichDockerReturnValue = 0
 
+    def cpe
+    def examinee;
+
     @Before
-    void setUp() {
-        super.setUp()
+    void init() {
 
         docker = new DockerMock()
         binding.setVariable('docker', docker)
         binding.setVariable('Jenkins', [instance: [pluginManager: [plugins: [new PluginMock()]]]])
 
-        echos = ''
-        helper.registerAllowedMethod("echo", [String.class], { String s -> echos += " $s" })
         helper.registerAllowedMethod('sh', [Map.class], {return whichDockerReturnValue})
+
+        prepareObjectInterceptors(this)
```

Comment:
do we really need this in this case?

Reply:
Yes. Otherwise the calls pointing to mocked steps (e.g. ```echo```) cannot be resolved.
Please cross check by removing/commenting the line above.